Add initial dhall templates for deployments

nholland94 opened the pull request:
![](https://media.giphy.com/media/yyvSeRGVj4C64/giphy.gif)

bkase commented on dhall/test.sh:
```diff
@@ -0,0 +1,25 @@
+END
+)
+
+dhall-to-yaml --documents "$@" < CodaNetwork.dhall
```
Can you add `--quoted` here? By default dhall-to-yaml admits the stupid yaml errors `on`, etc

bkase commented on dhall/External/Prelude.dhall:
```diff
@@ -0,0 +1,2 @@
+{- TODO: pick version tag and encode sha256 -}
+https://prelude.dhall-lang.org/package.dhall
```
recommend `v15.0.0` and then run `dhall freeze --inplace https://prelude.dhall-lang.org/v15.0.0/package.dhall` and it will inject the sha

nholland94 replied:
Yeah, was just holding off on pinning the version till I got things working. Gonna do that same for kubernetes as well.

nholland94 commented:
Took the weekend to bring the Dhall code up to date with the helm chart. There were a few minor changes made to the structure of the resulting yaml, which can be seen using the provided `diff_test.sh` (recommend installing `dyff` and running as `DIFF="dyff between" ./diff_test.sh` to have actually useful output).

I added an abstraction for representing arbitrary pod sidecars which can be "attached" to pod specs. Now you can specify a list of pod sidecar specs along with your coda node roles and it will build the specified sidecars and inject them into the coda node's pod spec. I also changed the `Component` pattern I was using to be compositional (as components should be). I wrote up an `README.md` that describes the `Component` pattern as well as the organization and import style standards of the project (thus far).

0x0I commented on dhall/README.md:
```diff
@@ -0,0 +1,107 @@
+
+## Imports
+
+As a standard, we maintain top-level directory structure in the qualified names of files we import. Only two directories in the project are special and do not follow this pattern: `Eternal/` and `Lib/`. Files in the `Lib/` folder do not have qualified names which are prefixed with `Lib/`, and same for `External/`. As a pattern, we tend to list imports 3 sections, each of which is ordered alphebet ...
```
nit: *alpha... -- also nice, this is looking good!

0x0I commented on dhall/Lib/List/filterMap.dhall:
```diff
@@ -0,0 +1,33 @@
+{- List/filterMap is a primitive which filters values from a list and replaces
+   the reamining items simultaneously. -}
```
nit: *remaining

yourbuddyconner commented:
For the record, we're going to revisit the overall structure of some of the components before officially merging this.
